src/model/operation/batchnorm.h

```diff
@@ -0,0 +1,62 @@
+//#ifndef SINGA_MODEL_OPERATION_BATCHNORM_H_
+//#define SINGA_MODEL_OPERATION_BATCHNORM_H_
+
+#include <vector>
+#include "singa/core/tensor.h"
+
+#ifdef USE_CUDNN
+#include <cudnn.h>
+#include "../layer/cudnn_utils.h" // check_cudnn
+#endif // USE_CUDNN 
+
+namespace singa {
+
+class BatchNormHandle {
+public:
+  BatchNormHandle(const float momentum, const Tensor& input, const Tensor& RunningMean, const Tensor& RunningVariance);
+
+  float factor;
+
+  size_t batchsize;
+  size_t channels;
+  size_t height;
+  size_t width;
+
+  Tensor runningMean;
```

Comment:
put runningMean and runningVariance at python level?

Reply:
runningMean and runningVariance should be in the same level with scale and bias. the only difference of them is that, runningMean and runningVariance don't need update. their date keep changing when do training. need to let batchnorm handle know the exist of them just like batchnorm layer know exist of them. their should not only be calculated gradually in training process, but also should support set initial value directly. export them to python helps pass init value from layer to handle.